ctapipe/calib/camera/r1.py

```diff
@@ -308,6 +312,127 @@ def real_calibrate(self, event):
             self.calibrator.ApplyEvent(samples, fci, r1)
 
 
+class _SST1M_BaselineFromRandomTrigger:
+    def __init__(self, n_bins):
+        self.n_bins = n_bins
+
+        self.n_events = None
+        self.means = []
+        self.stds = []
+
+        self.mean = None
+        self.std = None
+
+    def process(self, event):
+        for telescope_id in event.r0.tels_with_data:
+            r0 = event.r0.tel[telescope_id]
+            if self.n_events is None:
+                self.n_events = self.n_bins // r0.waveform.shape[1]
+
+            if r0.camera_event_type == 8:
+                self.means.append(r0.waveform.mean(axis=1))
+                self.means = self.means[-self.n_events:]
+                self.stds.append(r0.waveform.std(axis=1))
+                self.stds = self.stds[-self.n_events:]
+
+            if len(self.means) == self.self.n_events:
+                self.mean = np.mean(self.means, axis=0)
+                self.std = np.mean(self.stds, axis=0)
+        return event
+
+
+
+class SST1MR1Calibrator(CameraR1Calibrator):
+    '''
+    *  subtract a pixel-wise rough electronic + NSB noise pedestal value,
+        so that the values fluctuate around 0 counts
+        (and record it as MON or SVC data, so it can be un-applied
+        or quality-checked later)
+    *  apply a pixel-wise multiplicative gain correction factor to correct
+        for differences in camera uniformity
+    *  apply a scalar scale and shift the signal so that it efficiently fits
+        in a 16-bit signed integer value with the required dynamic range
+        (see Section 3.3)
+    *  record all applied corrections in SVC or MON streams.
+    '''
+
+    n_bins_for_baseline_estimation = Integer(1000).tag(config=True)
+    dark_baseline_path = Unicode(
+        "",
+        allow_none=True,
+        help='Path to a SST1M dark-baseline file (at the moment .npz)'
+    ).tag(config=True)
+
+    def __init__(self, config=None, tool=None, **kwargs):
+        super().__init__(config=config, tool=tool, **kwargs)
+        self.baseline_estimator = _SST1M_BaselineFromRandomTrigger(
+            n_bins=self.n_bins_for_baseline_estimation
+        )
+        if os.path.isfile(self.dark_baseline_path):
+            self.dark_baseline = np.load(self.dark_baseline_path)['baseline']
```

Comment:
Any files you load must be recorded in the provenance system, so either use `utils.get_dataset()` to do it automatically (which also searches automatically everywhere in `CTAPIPE_SVC_PATH`, which can contain multiple directories), or add the following line afterward:

```py3
Provenance().add_input_file(self.dark_baseline_path, role='dl0.tel.svc.sst1m_baseline')
```

Reply:
probably best to just use 
```py3
get_dataset(self.dark_baseline_path, role='dl0.tel.svc.sst1m_baseline')
```
since having an abstraction for reading files allows us to later switch to reading this from a database, or remote server without changing any code.